### PR TITLE
[CLI-46] prevent unloading remarks from being submitted multiple times without a response

### PR DESCRIPTION
**Pull request by ericohwotu**

### CODE REVIEW

File: app/controllers/actions/MessageTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.xml.NodeSeq
+
+class MessageTransformer @Inject()(implicit val executionContext: ExecutionContext) extends MessageTransformerInterface with Logging {
+
+  override def refine[A](request: ArrivalRequest[A]): Future[Either[Result, MessageTransformRequest[A]]] =
+    request.body match {
+      case x: NodeSeq =>
+        x.headOption.flatMap(node => messageResponse(request.channel)(node.label)) match {
```
Comment (MatthewAlexanderBeck) [inline app/controllers/actions/MessageTransformer.scala:46]: Might be worth double checking that the head node will always be the identifier. Not sure if api adds any wrappers etc.

File: app/models/ArrivalStatus.scala [hunk 2]
```diff
@@ -57,6 +56,7 @@ object ArrivalStatus extends Enumerable.Implicits with MongoDateTimeFormats {
     }
   }
 
+  /*TODO: investigate this as part of a separate ticket*/
```
Comment (MatthewAlexanderBeck) [inline app/models/ArrivalStatus.scala:59]: Can we remove this when the ticket is created as this is live code

File: test/controllers/actions/MessageTransformerSpec.scala [hunk 3]
```diff
@@ -33,63 +30,84 @@ import models.UnloadingRemarksRejectedResponse
...
 
           val request =
-            ArrivalRequest(FakeRequest("", "").withHeaders("X-Message-Type" -> MessageType.ArrivalRejection.code), arrivalMovement, channel)
+            ArrivalRequest(
+              FakeRequest("", "")
+                .withHeaders(
+                  "X-Message-Type" -> MessageType.ArrivalRejection.code,
+                  "Content-Type"   -> "application/xml"
+                )
+                .withBody[NodeSeq](<CC008A> </CC008A>),
```
Comment (MatthewAlexanderBeck) [inline test/controllers/actions/MessageTransformerSpec.scala:85]: Is it worth adding a Gen here for the rootnodes to test them all (we have them available in the MessageType model)

File: app/models/ArrivalStatus.scala (hunk 2 above)
Comment (ericohwotu) [reply app/models/ArrivalStatus.scala:59]: Removing

File: test/controllers/actions/MessageTransformerSpec.scala (hunk 3 above)
Comment (ericohwotu) [reply test/controllers/actions/MessageTransformerSpec.scala:85]: This also depends on the status of the arrival so might be a complex generator. will test all of them manually

**Review by MatthewAlexanderBeck — APPROVED** [review]

**Review by crystalmark — APPROVED** [review]

**Pull request merged**